### PR TITLE
curvefs client, metaserver : add refresh inode

### LINKED ISSUE #944: vebench test s3Compact cause data inconsistency
**Describe the bug (描述bug)**
image

**To Reproduce (复现方法)**

**Expected behavior (期望行为)**

**Versions (各种版本)**
OS:
Compiler:
curve-mds:
curve-chunkserver:
curve-snapshotcloneserver:
curve-sdk:
nebd:
curve-nbd:

**Additional context/screenshots (更多上下文/截图)**

### PR DESCRIPTION
**Pull request by xu-chaojie**
### What problem does this PR solve?

Issue Number: close #944  

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by h0hmj** [comment]
Sync() = UpdateInode+AppendS3ChunkInfo(false, not get inode in response)
Refresh() = UpdateInode+AppendS3ChunkInfo(true, get inode in response)

a liitle bit duplicate, but maybe it's ok.

File: curvefs/proto/metaserver.proto [hunk 1]
```diff
@@ -313,11 +313,13 @@ message AppendS3ChunkInfoRequest {
     required uint64 inodeId = 5;
     map<uint64, S3ChunkInfoList> s3ChunkInfoAdd = 6;
     map<uint64, S3ChunkInfoList> s3ChunkInfoRemove = 7;
+    required bool returnInode = 8;
```
Comment (ilixiaocui) [inline curvefs/proto/metaserver.proto:316]: GetOrModifyChunkInfoReuqest?

AppendS3ChunkInfoRequest  is confused. it is responsible for a lot of functionality, not just Append.

File: curvefs/src/client/s3/client_s3_cache_manager.cpp [hunk 2]
```diff
@@ -346,10 +346,11 @@ int FileCacheManager::Read(uint64_t inodeId, uint64_t offset, uint64_t length,
                     // ret -2 refs s3obj not exist
                     // clear inodecache && get again
                     LOG(INFO) << "inode cache maybe steal, try to get latest";
-                    inodeManager->ClearInodeCache(inodeId);
-                    auto r = inodeManager->GetInode(inodeId, inodeWrapper);
+                    ::curve::common::UniqueLock lgGuard =
+                        inodeWrapper->GetUniqueLock();
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/client_s3_cache_manager.cpp:350]: inodeManager->ClearInodeCache(inodeId) It is called in many places（fuse_client.cpp）, does it have the same problem？

Comment (xu-chaojie) [reply curvefs/src/client/s3/client_s3_cache_manager.cpp:350]: Other places is used to clear dir‘s inode， as so far， not cause any ploblem， we will fix it later.

File: curvefs/proto/metaserver.proto (hunk 1 above)
Comment (xu-chaojie) [reply curvefs/proto/metaserver.proto:316]: done

**Comment by xu-chaojie** [comment]
recheck

**Review by ilixiaocui — APPROVED** [review]

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**